File: Search in a Binary Search Tree/Search_in_a_Binary_Search_Tree.py
# Runtime: 84 ms, faster than 79.50% of Python3 online submissions for Search in a Binary Search Tree.
# Memory Usage: 15.2 MB, less than 5.30% of Python3 online submissions for Search in a Binary Search Tree.

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def searchBST(self, root: TreeNode, val: int) -> TreeNode:
        while root:
            if root.val == val:
                break
            elif root.val > val:
                root = root.right
            else:
                root = root.left;
        return root

File: Search in a Binary Search Tree/Search_in_a_Binary_Search_Tree.cpp
// Runtime: 60 ms, faster than 90.80% of C++ online submissions for Search in a Binary Search Tree.
// Memory Usage: 29.2 MB, less than 92.83% of C++ online submissions for Search in a Binary Search Tree.

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution 
{
public:
    TreeNode* searchBST(TreeNode* root, int val) 
    {
        while (root)
        {
            if (root->val == val)
                break;
            else if (root->val > val)
                root = root->left;
            else
                root = root->right;
        }
        return root;
    }
};